src/DonationForms/Rules/CurrencyRule.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Give\DonationForms\Rules;
+
+use Closure;
+use Give\Vendors\StellarWP\Validation\Contracts\ValidationRule;
+
+/**
+ * Custom currency validation rule that uses GiveWP's currency list and filter.
+ * Replaces the generic stellar validation library currency rule to ensure consistency
+ * with GiveWP's configured currencies and filters.
+ * 
+ * @unreleased
+ */
+class CurrencyRule implements ValidationRule
+{
+    /**
+     * @unreleased
+     */
+    public static function id(): string
+    {
+        return 'giveCurrency';
+    }
+
+    /**
+     * @unreleased
+     */
+    public static function fromString(string $options = null): ValidationRule
+    {
+        return new self();
+    }
+
+    /**
+     * Validates that the currency code is in GiveWP's supported currency list.
+     * Uses give_get_currencies_list() to get the current supported currencies
+     * and provides clear error messages with valid currency options.
+     * 
+     * @unreleased
+     */
+    public function __invoke($value, Closure $fail, string $key, array $values)
+    {
+        if (empty($value)) {
+            return;
+        }
+
+        // Get GiveWP's supported currencies
+        $supportedCurrencies = array_keys(give_get_currencies_list());
+
+        if (!in_array($value, $supportedCurrencies, true)) {
```

Comment:
I wonder if we should account for string case meaning our options are:
- a. keep the check for in_array which would imply exact uppercase syntax 
- b. uppercase the value before checking for in_array so the case does not matter

I'm inclined to keep what you have (a) because the Money class we use checks against the exact values without modifying the case & I believe we also have been storing these values as uppercase in the database.  Still would like your opinion 😄